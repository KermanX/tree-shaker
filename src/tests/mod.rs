use crate::{TreeShakeConfig, TreeShakeOptions};
use insta::{assert_snapshot, glob};
use oxc::{codegen::CodegenOptions, minifier::MinifierOptions};
use std::fs;

fn tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = crate::tree_shake(
    input,
    TreeShakeOptions {
      config: TreeShakeConfig::default().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions::default()),
      codegen_options: CodegenOptions::default(),
    },
  );
  result.codegen_return.code
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(tree_shake(input));
    })
  });
}
