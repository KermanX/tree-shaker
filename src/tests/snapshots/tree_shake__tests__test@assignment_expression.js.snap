---
source: src/tests/mod.rs
expression: tree_shake(input)
input_file: src/tests/fixtures/assignment_expression.js
---
export function f1() {
	return 2;
}
export function f2(__unused_E8B5) {
	return 2;
}
export function f3(a, b) {
	if (b) a = 2;
	return a;
}
export function f4(__unused_7C4F, __unused_9371) {
	return 3;
}
export function f5(a, b) {
	a = b;
	return a;
}
export function f6(a, b) {
	a.p = b;
	global.p = 1;
	effect;
	return a;
}
export function f7(a) {
	({x: __unused__, a: __unused__, b: {c: __unused__},...__unused__} = a);
	effect("a");
}
var __unused__;
